### ISSUE TITLE
GLEW Init Failed when using Remote Desktop

### ISSUE THREAD

**Issue by amenzies** [cnr-isti-vclab/meshlab#175]
In MeshLab_64bit_fp v2016.12 running meshlabserver.exe  while logged in through Windows Remote Desktop results in the error message "GLEW Init Failed!".  Running meshlab.exe fails to launch the GUI app and it looks like there is a popup notification dialog that disappears immediately.   Both programs work if an alternate remote viewer such as TeamViewer is used.

When using v1.3.3, meshlabserver and meshlab run correctly when run under Remote Desktop.  

Assuming this is because windows defaults to a generic graphics card driver when in remote desktop mode.

The specific filter script I've tested with is is `<filter name=""Simplification: Quadric Edge Collapse Decimation"">` but I assume this will occur with any filter.

Machine is Windows 10 Pro (64-bit), NVIDIA GeForce GTX 770, Intel I7.

Cheers,
-Alex

**Comment by harrysummer** [comment]
In Windows Remote Desktop, the session only has basic OpenGL 1.0 support. Therefore, all OpenGL-based application that requires modern OpenGL operations, such as GLSL, would fail to initialize. I think this is not an issue with MeshLab, but Windows. Maybe the only workaround is using a CPU implementation of OpenGL.

**Comment by paulpv** [comment]
Blender has a similar limitation that has a workaround by dropping an opengl32.dll in the same directory as blender.exe.
Maybe meshlab could do something similar?

**Comment by harrysummer** [comment]
@paulpv Thanks Paul! You really amazing! From your clues I searched and found:

https://blender.stackexchange.com/questions/50321/how-to-run-blender-through-rdp-on-windows-server-2008-r2-opengl-error

I tried with my remote desktop connected Win10 box with meshlab installed. And it succeed!! MeshLab opened and functioned smoothly.

I confirm and agree that the workaround "works" for MeshLab. However, whether or not to make it a part of official distribution of MeshLab is another problem because:
1. The workaround is a CPU implementation of opengl32.dll, which is part of other projects. License status unclear.
2. We need a mechanism to switch between GPU and CPU implementation. The possibility still remains a question. (I personally do not feel it possible, since Blender also only provides it as "workaround")
3. There is already [some place ][https://download.blender.org/ftp/sergey/softwaregl/ ] that can download the needed CPU-implemented DLL file. Why bother create another one?

**Comment by adrian-bodenmann** [comment]
On my computer Meshlab v1.3.3 does not work when run under Remote Desktop. 
I tried harrysummer's solution, which got me as far as opening a mesh, but I was unable to rotate it. 
I noticed there is a file opengl32sw.dll in the same folder as meshlab.exe and renaiming it into opengl32.dll solved the issue. So it looks like it is ready there, and I wonder why it cannot switch automatically to using opengl32sw.dll when started via remote desktop.
